File: server/src/weblab/data/dto/Permission.py
#!/usr/bin/python
# -*- coding: utf-8 -*-
#
# Copyright (C) 2005-2009 University of Deusto
# All rights reserved.
#
# This software is licensed as described in the file COPYING, which
# you should have received as part of this distribution.
#
# This software consists of contributions made by many individuals, 
# listed below:
#
# Author: Jaime Irurzun <jaime.irurzun@gmail.com>
#

class Permission(object):
    
    def __init__(self, name):
        super(Permission, self).__init__()
        self.name = name
        self.parameters = []
        
    def add_parameter(self, parameter):
        self.parameters.append(parameter)
        
    def __repr__(self):
        return "Permission(name = '%s', parameters = '%r')" % (
                self.name,
                self.parameters
            )
        

class PermissionParameter(object):
    
    def __init__(self, name, datatype, value):
        super(PermissionParameter, self).__init__()
        self.name = name
        self.datatype = datatype
        self.value = value   
        
    def __repr__(self):
        return "PermissionParameter(name = '%s', datatype = '%s', value = '%s')" % (
                self.name,
                self.datatype,
                self.value
            )

File: gwt-client/src/es/deusto/weblab/client/dto/users/Permission.java
/*
* Copyright (C) 2005 onwards University of Deusto
* All rights reserved.
*
* This software is licensed as described in the file COPYING, which
* you should have received as part of this distribution.
*
* This software consists of contributions made by many individuals, 
* listed below:
*
* Author: Jaime Irurzun <jaime.irurzun@gmail.com>
*
*/

package es.deusto.weblab.client.dto.users;

public class Permission {
	
	private String name;
	private PermissionParameter[] parameters;
	
	public Permission(String name, PermissionParameter[] parameters) {
		super();
		this.name = name;
		this.parameters = parameters;
	}
	
	public String getName() {
		return this.name;
	}
	public void setName(String name) {
		this.name = name;
	}
	public PermissionParameter[] getParameters() {
		return this.parameters;
	}
	public void setParameters(PermissionParameter[] parameters) {
		this.parameters = parameters;
	}
	
	@Override
	public boolean equals(Object other) {
		if ( other instanceof Permission ) {
			if ( this.name != ((Permission)other).name )
				return false;
			if ( this.parameters.length != ((Permission)other).parameters.length )
				return false;
			for ( int i = 0; i < this.parameters.length; ++i )
				if ( this.parameters[i] != ((Permission)other).parameters[i] )
					return false;
		} else {
			return false;
		}
		return true;
	}
}
